agent/src/protocols/mod.rs

```diff
@@ -8,7 +8,7 @@ use blake2::VarBlake2b;
 use failure::Error;
 use std::collections::HashMap;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
maybe add a unit test to validate:

```rust
    #[tokio::test]
    async fn test_discovery_result_partialeq() {
        let left = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
        let right = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
        assert_eq!(left, right);
   }

    #[tokio::test]
    async fn test_discovery_result_partialeq_false() {
        {
            let left = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
            let right = DiscoveryResult::new(&"foo2".to_string(), HashMap::new(), true);
            assert_ne!(left, right);
        }

        {
            let left = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
            let right = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), false);
            assert_ne!(left, right);
        }

        {
            let mut nonempty = HashMap::new();
            nonempty.insert("one", "two");
            let left = DiscoveryResult::new(&"foo1".to_string(), nonempty, true);
            let right = DiscoveryResult::new(&"foo1".to_string(), HashMap::new(), true);
            assert_ne!(left, right);
        }
   }
```

Reply:
I may need some guidance on your environment variable mocker (see line 186)

Fixed: https://github.com/deislabs/akri/pull/176/commits/63c4bbbe9e690036e012c66dbacb727efa290c89

Reply:
interesting ... i didn't see that coming, lol.  

to do this right, we would need to essentially modify `DiscoveryResult::new` to accept `query: &impl EnvVarQuery` as a parameter (like `inner_get_discovery_handler`), then use query in `DiscoveryResult::new`.  in the test, we'd use a `MockEnvVarQuery` like `test_factory_for_debug_echo_when_no_env_var_set()`.

but that might be more of a change than you are interested in making :)

another possibility would be to remove that `DisoceryResult::new(..., false)` test (or comment it out with a note saying some work needs to be done to `DiscoveryResult::new` to enable the test)

Reply:
If ok with you, I'll exclude the `false` shared for now with a comment and then, perhaps next week (?), I'll learn what you've explained and submit a follow-up PR?

Reply:
Per discussion: https://github.com/deislabs/akri/pull/176/commits/1b02e672b49851fde7d6bce26f310021839b5193